pkgs/by-name/ca/cargo-tauri/package.nix

```
@@ -4,39 +4,41 @@
   callPackage,
   rustPlatform,
   fetchFromGitHub,
+  cargo-tauri,
   darwin,
-  gtk3,
-  libsoup,
+  gtk4,
+  nix-update-script,
   openssl,
   pkg-config,
-  webkitgtk,
+  testers,
+  webkitgtk_4_1,
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "tauri";
-  version = "1.7.1-unstable-2024-08-16";
+  version = "2.0.1";
 
   src = fetchFromGitHub {
     owner = "tauri-apps";
     repo = "tauri";
-    rev = "2b61447dfc167ec11724f99671bf9e2de0bf6768";
-    hash = "sha256-gKG7olZuTCkW+SKI3FVZqgS6Pp5hFemRJshdma8rpyg=";
+    rev = "refs/tags/tauri-v${version}";
+    hash = "sha256-kHzpx1o894bo4Ud3D1lbB+pZahflF/o2bb2EDJUI148=";
   };
 
-  # Manually specify the sourceRoot since this crate depends on other crates in the workspace. Relevant info at
-  # https://discourse.nixos.org/t/difficulty-using-buildrustpackage-with-a-src-containing-multiple-cargo-workspaces/10202
-  sourceRoot = "${src.name}/tooling/cli";
-
-  cargoHash = "sha256-VXg/dAhwPTSrLwJm8HNzAi/sVF9RqgpHIF3PZe1LjSA=";
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

```suggestion
    lockFile = ${src}/Cargo.lock;
```
As from comment https://github.com/NixOS/nixpkgs/pull/346277/files#r1788374298

---

This suggestion works.

---

This is IFD, which is [not allowed in nixpkgs](https://github.com/NixOS/nixpkgs/tree/92248b23e5925b7a69108488e1f6d4577eea9f76/pkgs#import-from-derivation)

---

Oh didn't know.